spec/graphql/pagination/array_connection_spec.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+require "spec_helper"
+
+describe GraphQL::Pagination::ArrayConnection do
+  class TestSchema < GraphQL::Schema
+    ITEMS = [
+      { name: "Avocado" },
+      { name: "Beet" },
+      { name: "Cucumber" },
+      { name: "Dill" },
+      { name: "Eggplant" },
+      { name: "Fennel" },
+      { name: "Ginger"},
+      { name: "Horseradish" },
+      { name: "I Can't Believe It's Not Butter" },
+      { name: "Jicama" },
+    ]
+
+    class Item < GraphQL::Schema::Object
+      field :name, String, null: false
+    end
+
+    class Query < GraphQL::Schema::Object
+      # TODO don't require `connection: false`
+      field :items, Item.connection_type, null: false, connection: false do
+        argument :first, Integer, required: false
+        argument :last, Integer, required: false
+        argument :before, String, required: false
+        argument :after, String, required: false
+      end
+
+      def items(**args)
+        GraphQL::Pagination::ArrayConnection.new(ITEMS, context, max_page_size: 6, **args)
```

Comment:
by making it an extension applied to the field definition, the resolver method here just becomes `ITEMS` and the extension can do the post-processing in its regular hook instead.